### DISCUSSION TITLE
Configurable damage of harmful blocks

### DISCUSSION THREAD

**Discussion by AllayBreeze** [PurpurMC/Purpur#1159] [__Dismissed__] [answered]
The ability to change the damage amount or interval dealt by some blocks such as cactus or magma block, or disable the damage at all.

The config:

    cactus:
      deal-damage: true
      damage-amount: 1
      damage-interval: 1
    campfire:
      deal-damage: true
      damage-amount: 1
      damage-interval: 1
    soul_campfire:
      deal-damage: true
      damage-amount: 2
      damage-interval: 1
    fire:
      deal-damage: true
      damage-amount: 1
      damage-interval: 1
    soul_fire:
      deal-damage: true
      damage-amount: 2
      damage-interval: 1
    lava:
      deal-damage: true
      damage-amount: 4
      damage-interval: 1
    lava_cauldron:
      deal-damage: true
      damage-amount: 4
      damage-interval: 1
    magma_block:
      deal-damage: true
      damage-amount: 1
      damage-interval: 1
    powder_snow:
      deal-damage: true
      damage-amount: 1
      damage-interval: 40
    powder_snow_cauldron:
      deal-damage: false
      damage-amount: 1
      damage-interval: 40
    sweet_berry_bush:
      deal-damage: true
      damage-amount: 1
      damage-interval: 1
    wither_rose:
      apply-wither-effect: true
      effect-interval: 1

**Comment by RhythmicSys** [comment, marked as answer]
1.19.4 has added the ability to do this through a datapack using the `damage_type` tag, so this will be better as a datapack, very soon lol
